lib/galaxy/schema/groups.py

```diff
@@ -0,0 +1,80 @@
+from typing import List
+
+from pydantic import Field
+from typing_extensions import Literal
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    ModelClassField,
+)
+from galaxy.schema.schema import Model
+
+GROUP_MODEL_CLASS = Literal["Group"]
+
+
+class GroupIndexResponse(Model):
+    """Response schema for a group."""
+
+    model_class: GROUP_MODEL_CLASS = ModelClassField(GROUP_MODEL_CLASS)
+    id: DecodedDatabaseIdField = Field(
+        ...,
+        title="group ID",
+    )
+    url: str = Field(
+        ...,
+        title="URL for the group",
+    )
+    name: str = Field(
+        ...,
+        title="name of the group",
+    )
+
+
+class GroupIndexListResponse(Model):
+    """Response schema for listing groups."""
+
+    __root__: List[GroupIndexResponse]
+
+
+class GroupShowResponse(Model):
```

Comment:
This model name seems very coupled with the "show" endpoint, but it can be used in other endpoints like the "create".

Reply:
I'm a bit confused. Isn't `GroupShowResponse` exclusively for the 'show' endpoint because of `roles_url` and `users_url`?